client/app/controllers/project.js

```diff
@@ -0,0 +1,12 @@
+import Controller from '@ember/controller';
+import { sort } from '@ember/object/computed';
+import { optionset } from '../helpers/optionset';
+
+export default class ProjectController extends Controller {
+
+  get publicStatusNotNullVisibilityGeneralPublic() {
+    const publicStatus = optionset(['project', 'dcpPublicstatus', 'label', this.model.dcpPublicstatus]);
```

Comment:
instead of getting the label, you can just directly match against the codes. 
```
const isGeneralPublic = this.model.dcpVisibility === optionset(['project', 'dcpVisibility', 'code', 'GENERAL_PUBLIC']);

return this.model.publicStatus && isGeneralPublic;
```